sql-cli/src/sql_cli/dag_generator.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from networkx import DiGraph, depth_first_search, find_cycle, is_directed_acyclic_graph
```

Comment:
Huge +1 for having an external library handle graph creation. We should scale test this with some different configurations to make sure there's no efficiency issues but otherwise I'm always a fan of offloading work.